File: Python/parse/lambda_function.py
import json

def parse(in_str):
    counts = [0] * 26

    for i in range(0, len(in_str)):
        val = ord(in_str[i])
        if val > 96 and val < 123:
            counts[val - 97] += 1

    return counts


def lambda_handler(event, context):
    # TODO implement
    size = event['size']
    data = open(f'string_input_{size}.json')
    return {
        'statusCode': 200,
        'body': parse(json.load(data)['data'])
    }

File: Ruby/parse/lambda_function.rb
require 'json'

def parse(in_str)
    counts = Array.new(26, 0)

    for i in 0..(in_str.length - 1) do
        val = in_str[i].ord
        if val > 96 && val < 123
            counts[val - 97] += 1
        end
    end
    return counts
end

def lambda_handler(event:, context:)
    filename = "string_input_%d.json" % [event['size']]
    file = File.read(filename)
    text = JSON.parse(file)['data']
    { statusCode: 200, body: parse(text) }
end
